src/components/DnsSettingsForm.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) [2020] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import {
+    FormGroup,
+    TextInput
+} from '@patternfly/react-core';
+import { useNetworkDispatch, updateDnsSettings } from '../context/network';
+import ModalForm from './ModalForm';
+import IPInput from './IPInput';
+
+const _ = cockpit.gettext;
+
+const DnsSettingsForm = ({ isOpen, onClose, dns }) => {
+    const dispatch = useNetworkDispatch();
+    const originalNameServers = (dns?.nameServers || "").split(" ");
+    const [nameserver1, setNameserver1] = useState(originalNameServers[0] || "");
+    const [nameserver2, setNameserver2] = useState(originalNameServers[1] || "");
+    const [nameserver3, setNameserver3] = useState(originalNameServers[2] || "");
+    const [policy, setPolicy] = useState(dns?.policy || "");
+    const [searchList, setSearchList] = useState(dns?.searchList || "");
+
+    const nameServers = () => {
+        return [nameserver1, nameserver2, nameserver3].filter(Boolean);
+    };
+
+    const configChanged = () => {
+        return ((nameServers().join(" ") !== (dns?.nameServers || "")) ||
```

Comment:
That makes me reaffirm in having a _more defined_ `dns` prop in this component, to avoid falling a "default" value here multiple times.

Reply:
yep, I started working in the form with a const and did all the fallbacks here which were later done in the initialization of the dnsSettings, so, will avoid remove it from the component.